compiler/modules/nback/debug.bal

```diff
@@ -0,0 +1,225 @@
+import wso2/nballerina.bir;
+import wso2/nballerina.print.llvm;
+import wso2/nballerina.types as t;
+import wso2/nballerina.comm.err;
+
+type DIBuilder llvm:DIBuilder;
+type DISubprogram llvm:Metadata;
+type DILocation llvm:Metadata;
+type DIFile llvm:Metadata;
+type DICompileUnit llvm:Metadata;
+type DISubroutineType llvm:Metadata;
+
+type Scope record {|
+    llvm:Metadata diScope;
+    bir:Position startPos;
+    bir:Position endPos;
+    Scope[] childScopes;
+|};
+
+type ModuleDI record {|
+    DIBuilder builder;
+    DIFile[] files;
+    DICompileUnit compileUnit;
+    DISubroutineType funcType;
+    boolean debugFull;
+|};
+
+// Debug location will always be added
+public const int DEBUG_USAGE_ERROR_CONSTRUCT = 0;
+public const int DEBUG_USAGE_CALL = 1;
+
+public type DebugLocationUsage DEBUG_USAGE_ERROR_CONSTRUCT|DEBUG_USAGE_CALL;
+
+class DIScaffold {
```

Comment:
This is done in rather OO, Java-ish style. Can you make stylistically consistent with Scaffold? The class should just mininimally encapsulate the state.  Most of the real work is done by module level functions not methods.

Reply:
fixed